### PR TITLE
5.7 rc1 rn

### PR DESCRIPTION
**Pull request by azhebel**

### CODE REVIEW

File: doc/source/release-notes/Percona-XtraDB-Cluster-5.7.12-rc1-26.16.rst [hunk 1]
```diff
@@ -0,0 +1,97 @@
+.. rn:: 5.7.12-rc1-26.16
+
+=========================================
+|Percona XtraDB Cluster| 5.7.12-rc1-26.16 
+=========================================
+
+Percona is glad to announce the release of
+|PXC| 5.7.12-rc1-26.16 on Audust 8, 2016.
```
Comment (hrvojem) [inline doc/source/release-notes/Percona-XtraDB-Cluster-5.7.12-rc1-26.16.rst]: typo: Audust

File: doc/source/release-notes/Percona-XtraDB-Cluster-5.7.12-rc1-26.16.rst [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+* [PXC-564] Added support for sending the keyring when performing encrypted SST.
+  For more information, see :ref:`encrypt-traffic`.
+
+* [PXC-571] Changed the code of ``THD_PROC_INFO``
+  to reflect what the thread is currently doing.
+
+* [PXC-612] Using XtraBackup as the SST method
+  now requires Percona XtraBackup 2.4.3 or later.
+
```
Comment (hrvojem) [inline doc/source/release-notes/Percona-XtraDB-Cluster-5.7.12-rc1-26.16.rst]: IIRC it should be 2.4.4

File: doc/source/release-notes/Percona-XtraDB-Cluster-5.7.12-rc1-26.16.rst [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+* [PXC-648] Removed the ``sst_special_dirs`` variable.
+
+* [PXC-659] Disabled switching of ``slave_preserve_commit_order`` to ``ON``.
+
+* [PQA-115] Changed the default :file:`my.cnf` configuration.
+
+---
+
+* [BLD-464] 455-458, 460, 461 should I mention BLD issues?
+
```
Comment (hrvojem) [inline doc/source/release-notes/Percona-XtraDB-Cluster-5.7.12-rc1-26.16.rst]: No, none of them are connected with beta release so no users have been affected.

File: doc/source/release-notes/Percona-XtraDB-Cluster-5.7.12-rc1-26.16.rst [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  or the ``--pxc-strict-mode`` option during ``mysqld`` startup.
+  For more information, see :ref:`pxc-strict-mode`.
+
+* [PXC-626] **Galera instruments exposed in Performance Schema**:
+  This includes mutexes, condition variables, file instances, and thread.
+
+Bug Fixes
+=========
+
+* [PXC-233] Fixed error messages.
```
Comment (kbauskar) [inline doc/source/release-notes/Percona-XtraDB-Cluster-5.7.12-rc1-26.16.rst]: Please re-check the PXC issue number.

https://jira.percona.com/browse/PXC-233 (Discuss hanging issues with Philllip)
Wasn't part of this release. It is already closed too.

File: doc/source/release-notes/Percona-XtraDB-Cluster-5.7.12-rc1-26.16.rst [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+* [PXC-585] Removed protection against repeated calls of ``wsrep->pause()``
+  on the same node to allow parallel RSU operation.
+
+* [PXC-592] Changed ``wsrep_row_upd_check_foreign_constraints``
+  to ensure that ``fk-reference-table`` is open before marking it open.
+
+* [PXC-601] Fixed error when running ``SHOW STATUS`` during group state update.
+
+* [PXC-605] Changed the ``sst_flush_tables()`` function
+  to return a non-negative error code and thus pass assertion.
```
Comment (kbauskar) [inline doc/source/release-notes/Percona-XtraDB-Cluster-5.7.12-rc1-26.16.rst]: Please update the description too:

"Corrected return code of sst_flush_tables()"

File: doc/source/release-notes/Percona-XtraDB-Cluster-5.7.12-rc1-26.16.rst [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+* [PXC-613] Fixed memory leak due to stats not freeing
+  when toggling the :variable:`wsrep_provider` variable.
```
Comment (kbauskar) [inline doc/source/release-notes/Percona-XtraDB-Cluster-5.7.12-rc1-26.16.rst]: s/memory_leak/memory leak and stale pointer/gc

File: doc/source/release-notes/Percona-XtraDB-Cluster-5.7.12-rc1-26.16.rst [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+* [PXC-624] Fixed ``galera_bf_abort`` failures by ensuring
```
Comment (kbauskar) [inline doc/source/release-notes/Percona-XtraDB-Cluster-5.7.12-rc1-26.16.rst]: You can get rid of this item. Internal TC fix.

File: doc/source/release-notes/Percona-XtraDB-Cluster-5.7.12-rc1-26.16.rst [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+  that nodes form a cluster.
+
+* [PXC-625] Fixed failure of ``ROLLBACK`` to register ``wsrep_handler``
+
+* [PXC-636] Fixed failure of symmetric encryption during SST.
+
+---
+
+* [PXC-630] [PXC-641] [PXC-647] [PXC-617] [PXC-650] very low-level?
```
Comment (kbauskar) [inline doc/source/release-notes/Percona-XtraDB-Cluster-5.7.12-rc1-26.16.rst]: No need to mention them.

File: doc/source/release-notes/Percona-XtraDB-Cluster-5.7.12-rc1-26.16.rst [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+
+* [PXC-587] [PXC-614] [PXC-616] [PXC-632] [PXC-639] [PXC-658] Fixed as part of pxc_strict_mode?
+
```
Comment (kbauskar) [inline doc/source/release-notes/Percona-XtraDB-Cluster-5.7.12-rc1-26.16.rst]: Same here. No need to mention them.

File: doc/source/release-notes/Percona-XtraDB-Cluster-5.7.12-rc1-26.16.rst [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+* [PXC-635] Error message fix. Need to mention?
+
```
Comment (kbauskar) [inline doc/source/release-notes/Percona-XtraDB-Cluster-5.7.12-rc1-26.16.rst]: Same here. No need to mention this one too.

File: doc/source/release-notes/Percona-XtraDB-Cluster-5.7.12-rc1-26.16.rst (hunk 2 above)
Comment (kbauskar) [inline doc/source/release-notes/Percona-XtraDB-Cluster-5.7.12-rc1-26.16.rst]: Right. PXC-5.7 needs 2.4.4.

File: doc/source/release-notes/Percona-XtraDB-Cluster-5.7.12-rc1-26.16.rst [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+* [PXC-612] Using XtraBackup as the SST method
+  now requires Percona XtraBackup 2.4.3 or later.
+
+* [PXC-628] [PXC-651] Improved rollback process to ensure that when a transaction
+  is rolled back, any statements open by the transaction are also rolled back.
+
+* [PXC-648] Removed the ``sst_special_dirs`` variable.
+
+* [PXC-659] Disabled switching of ``slave_preserve_commit_order`` to ``ON``.
```
Comment (kbauskar) [inline doc/source/release-notes/Percona-XtraDB-Cluster-5.7.12-rc1-26.16.rst]: append
"when running PXC in cluster-mode as it conflicts with Galera existing multi-master commit ordering resolution algorithm"

File: doc/source/release-notes/Percona-XtraDB-Cluster-5.7.12-rc1-26.16.rst [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+
+* [PQA-115] Changed the default :file:`my.cnf` configuration.
+
+---
+
+* [BLD-464] 455-458, 460, 461 should I mention BLD issues?
+
+* [PXC-267] [PXC-268] Do these change anything for users?
+
```
Comment (kbauskar) [inline doc/source/release-notes/Percona-XtraDB-Cluster-5.7.12-rc1-26.16.rst]: You can skip this.

and may be add a generic statement

"and more stability fixes" if that make sense and followed practice.

File: doc/source/release-notes/release-notes_index.rst [hunk 13]
```diff
@@ -6,5 +6,6 @@
    :maxdepth: 1
    :glob:
 
-   Percona-XtraDB-Cluster-5.7.11-4beta-25.14.2
+   Percona-XtraDB-Cluster-5.7.12-rc1-26.16
```
Comment (kbauskar) [inline doc/source/release-notes/release-notes_index.rst]: Ideally name should be -5rc1 and not -rc1.
Do you check this with Evgeniy.

I see this in version.
[kbauskar@kbauskar-Inspiron-7548 percona-xtradb-cluster] $cat VERSION 
MYSQL_VERSION_MAJOR=5
MYSQL_VERSION_MINOR=7
MYSQL_VERSION_PATCH=12
MYSQL_VERSION_EXTRA=-5rc1

**Pull request merged**